### DISCUSSION TITLE
GLIBC版本问题

### DISCUSSION THREAD

**Discussion by bigbao9494** [xmake-io/xmake#3417] [Q&A]
环境：ubuntu20，xmake-v2.7.7.amd64.deb
升级过了glibc：
strings /lib/x86_64-linux-gnu/libc.so.6 |grep GLIBC_
GLIBC_2.2.5
GLIBC_2.2.6
GLIBC_2.3
GLIBC_2.3.2
GLIBC_2.3.3
GLIBC_2.3.4
GLIBC_2.4
GLIBC_2.5
GLIBC_2.6
GLIBC_2.7
GLIBC_2.8
GLIBC_2.9
GLIBC_2.10
GLIBC_2.11
GLIBC_2.12
GLIBC_2.13
GLIBC_2.14
GLIBC_2.15
GLIBC_2.16
GLIBC_2.17
GLIBC_2.18
GLIBC_2.22
GLIBC_2.23
GLIBC_2.24
GLIBC_2.25
GLIBC_2.26
GLIBC_2.27
GLIBC_2.28
GLIBC_2.29
GLIBC_2.30
GLIBC_2.31   //以下为新增
GLIBC_2.32
GLIBC_2.33
GLIBC_2.34
GLIBC_2.35
GLIBC_PRIVATE

支行xmake：
xmake: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by xmake)
xmake: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by xmake)
xmake: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by xmake)

**Comment by waruqi** [comment]
不要用 xmake-v2.7.7.amd64.deb 包安装，那个是 ci 上预构建的，可能会依赖ci上的 glibc 环境，后面我干脆 releases 删了  deb 包好了，避免误用。


官网文档和 README 上都是推荐一键安装脚本，简单直接。

```
bash <(curl -fsSL https://xmake.io/shget.text)
```

也可以走 releases 下  xmake-xxx.gz.run 自安装脚本，也是本地编译安装，也可以。。

再不行就源码编译安装，这里有完整安装文档：https://xmake.io/#/zh-cn/guide/installation

**Comment by waruqi** [comment]
releases 上的 deb 包我删了，以后还是统一安装方式，走脚本装

**Reply by bigbao9494** [reply]
@waruqi 非常感谢。。
我还有另外一个问题，在WINDOWS下生成VS2019工程
在WINDOWS中DEBUG版本时始终无法成功下载openssl，尝试了很多次都失败

xmake f --with_libcurl=true --with_openssl=true
xmake project -k vsxmake -a "x64,x86" -m "debug" vs

**Reply by waruqi** [reply]
下载问题看下这里 https://xmake.io/#/zh-cn/package/remote_package?id=%e8%bf%9c%e7%a8%8b%e5%8c%85%e4%b8%8b%e8%bd%bd%e4%bc%98%e5%8c%96
